core/src/main/java/org/sonarsource/sonarlint/core/container/global/ExtensionInstaller.java

```diff
@@ -76,6 +67,29 @@ public ExtensionInstaller install(ComponentContainer container) {
     return this;
   }
 
+  private static void loadExtensions(ComponentContainer container, PluginInfo pluginInfo, Plugin.Context context) {
+    for (Object extension : context.getExtensions()) {
+      Boolean isSlPluginOrNull = pluginInfo.isSonarLintSupported();
+      boolean isExplicitlySonarLintCompatible = isSlPluginOrNull != null && isSlPluginOrNull.booleanValue();
+      if (isExplicitlySonarLintCompatible) {
+        // When plugin itself claim to be compatible with SonarLint, only load @SonarLintSide extensions
+        // filter out non officially supported Sensors
+        if (ExtensionUtils.isSonarLintSide(extension) && (PluginCopier.isWhitelisted(pluginInfo.getKey()) || isNotSensor(extension))) {
+          container.addExtension(pluginInfo, extension);
+        }
+      } else if (!blacklisted(extension) && (ExtensionUtils.isScannerSide(extension) || ExtensionUtils.isType(extension, ProfileDefinition.class))) {
+        // Here we have whitelisted extensions of whitelisted plugins
+        container.addExtension(pluginInfo, extension);
+      } else {
+        LOG.debug("Extension {} was blacklisted as it is not used by SonarLint", className(extension));
+      }
+    }
+  }
+
+  private static boolean isNotSensor(Object extension) {
+    return !ExtensionUtils.isType(extension, Sensor.class) && !ExtensionUtils.isType(extension, org.sonar.api.batch.Sensor.class);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "Sensor"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)